pkg/kubelet/kubelet.go

```diff
@@ -2066,19 +2066,28 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	sctx := context.WithoutCancel(ctx)
 	result := kl.containerRuntime.SyncPod(sctx, pod, podStatus, pullSecrets, kl.crashLoopBackOff)
 	kl.reasonCache.Update(pod.UID, result)
-	if err := result.Error(); err != nil {
-		// Do not return error if the only failures were pods in backoff
-		for _, r := range result.SyncResults {
-			if r.Error != kubecontainer.ErrCrashLoopBackOff && r.Error != images.ErrImagePullBackOff {
-				// Do not record an event here, as we keep all event logging for sync pod failures
-				// local to container runtime, so we get better errors.
-				return false, err
+
+	returnError := false
+	for _, r := range result.SyncResults {
+		if r.Action == kubecontainer.ResizePod {
+			if r.Message == "" {
+				// The pod was resized successfully, clear any pod resize errors in the PodResizeInProgress condition.
+				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", true)
+			} else {
+				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, v1.PodReasonError, r.Message, false)
 			}
 		}
-
-		return false, nil
+		// Do not return error if the only failures were pods in backoff
+		if r.Error != kubecontainer.ErrCrashLoopBackOff && r.Error != images.ErrImagePullBackOff {
+			// Do not record an event here, as we keep all event logging for sync pod failures
+			// local to container runtime, so we get better errors.
+			returnError = true
+		}
 	}
 
+	if returnError {
+		return false, result.Error()
+	}
 	return false, nil
```

Comment:
Just because one container is in crashloopbackoff doesn't mean that none of the other error results are relevant. The check is broken anyway (https://github.com/kubernetes/kubernetes/issues/123602). I think we should just remove the crashloopbackoff block, and just return `result.Error()`, no need for `returnError` tracking.

/cc @hankfreund

```suggestion
	}
	return false, result.Error()
```

Reply:
> This code is non-functional

I felt like I was mildly insane when I was going through this code and trying to understand how it worked... good to know that the answer is that it doesn't 😊